session/session.go

```diff
@@ -271,20 +273,21 @@ func (s *Session) subscribe(dc *webrtc.DataChannel) {
 	})
 }
 
-func (s *Session) Attach(path string, args ...string) error {
-	cmd := exec.CommandContext(s.ctx, path, args...)
+func (s *Session) Run(path string, args ...string) error {
+	done := errors.New("done")
+
+	ctx, cancel := context.WithCancelCause(s.ctx)
+	cmd := exec.CommandContext(ctx, path, args...)
```

Comment:
Regardless of whether this PR in its current form ends up in `main` or not, we definitely should make use of a child context for managing the attached command's lifespan. `main` in it's current state does not.